Tests/src/InternalTests/EncodeTests.cpp

```diff
@@ -0,0 +1,50 @@
+/*
+ * Copyright 2023 Magnopus LLC
+
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#ifndef SKIP_INTERNAL_TESTS
+
+#include "TestHelpers.h"
+#include "Common/Encode.h"
+
+#include "gtest/gtest.h"
+
+CSP_INTERNAL_TEST(CSPEngine, EncodeTests, EncodeURI)
```

Comment:
As we are using double encoding for the sequence keys I think our tests should cover that.